nyxx.interactions/lib/src/Interactions.dart

```diff
@@ -26,42 +32,58 @@ class Interactions {
     });
   }
 
-  Future<SlashCommand> registerSlashGlobal(
-      String name, String description, List<SlashArg> args) async {
-    final command =
-        SlashCommand._new(this._client, name, args, description, null);
-    final response = await this._client.httpEndpoints.sendRawRequest(
-          "/applications/${this._client.app.id.toString()}/commands",
-          "POST",
-          body: command._build(),
-        );
+  SlashCommand createCommand(String name, String description, List<CommandArg> args, {String? guild})
+    => SlashCommand._new(_client, name, description, args, guild: guild != null ? CacheUtility.createCacheableGuild(_client, Snowflake(guild)) : null);
 
-    if (response is HttpResponseError) {
-      return Future.error(response);
-    }
-
-    return Future.value(command);
+  /// Registers a single command.
+  ///
+  /// @param command A [SlashCommand] to register.
+  void registerCommand(SlashCommand command) {
+    _commands.add(command);
   }
 
-  Future<SlashCommand> registerSlashGuild(String name, String description,
-      Snowflake guildId, List<SlashArg> args) async {
-    final command =
-        SlashCommand._new(this._client, name, args, description, guildId);
-    command._build();
-    final response = await this._client.httpEndpoints.sendRawRequest(
-          "/applications/${this._client.app.id.toString()}/guilds/${guildId.toString()}/commands",
-          "POST",
-          body: command._build(),
-        );
+  /// Registers multiple commands at one.
+  ///
+  /// This wraps around [Interactions.registerCommand()] running [Interactions.registerCommand] for each command in the list.
+  /// @param commands A list of [SlashCommand]s to register.
+  void registerCommands(List<SlashCommand> commands) {
+    for(var i = 0; i < commands.length; i++) {
+      this.registerCommand(commands[i]);
+    }
+  }
 
-    if (response is HttpResponseError) {
-      return Future.error(response);
+  List<SlashCommand> getCommands({bool registeredOnly = false}) {
+    if(!registeredOnly) { return _commands; }
+    final registeredCommands = List<SlashCommand>.empty(growable: true);
+    for(var i = 0; i < _commands.length; i++) {
+      final el = _commands[i];
+      if(el.isRegistered) {
+        registeredCommands.add(el);
+      }
     }
+    return registeredCommands;
+  }
 
-    return Future.value(command);
+  Future<void> sync() async {
+    var success = 0;
+    var failed = 0;
+    for(var i = 0; i < _commands.length; i++) {
+      final el = _commands[i];
+      if(!el.isRegistered) {
+        final command = await el._register();
+        if(command != null) {
+          _commands[i] = command;
+          this._events.onSlashCommandCreated.add(_commands[i]);
+          success++;
+        } else {
+          failed++;
+        }
+      }
+    }
+    _logger.info("Successfully registered $success ${success > 1 ? "commands" : "command"}. Failed registering $failed ${failed > 1 ? "commands" : "command"}.");
```

Comment:
Is that whole logic needed? Command can fail to register?

Reply:
I have it there in case of a failure, not 100% sure but that number will be incremented in the case of an error being caught while registering.